qiita_db/metadata_template/test/test_sample_template.py

```diff
@@ -1026,55 +977,31 @@ def test_create_int_prefix(self):
                 'sample_type': "type1",
                 'str_column': "Value for sample 3",
                 'taxon_id': 9606,
-                'scientific_name': 'homo sapiens'}]
-        self.assertEqual(obs, exp)
+                'scientific_name': 'homo sapiens'}}
+        for s_id in exp_sample_ids:
+            self.assertEqual(st[s_id]._to_dict(), exp_dict[s_id])
 
     def test_create_str_prefixes(self):
         """Creates a new SampleTemplate with sample names string prefixed"""
-        new_id = self.new_study.id
         st = SampleTemplate.create(self.metadata_str_prefix, self.new_study)
+        new_id = self.new_study.id
```

Comment:
The previous test failed, because now in the `qiita.study_sample` table we are storing more information, but just the link between the study and the samples. This made the line 1045 (old code) to fail. Following @wasade's proposal of making the tests independent of the storage system, I change the tests to be using the objects API. This way, if in the future we want to change the storage technology, we will not need to modify the test.

However, I just noticed that I'm missing here the 2 new properties. Will add.